src/Symfony/Component/Messenger/Event/WorkerMessageReceivedEvent.php

```diff
@@ -28,4 +30,9 @@ public function shouldHandle(bool $shouldHandle = null): bool
 
         return $this->shouldHandle;
     }
+
+    public function setEnvelope(Envelope $envelope): void
+    {
+        $this->envelope = $envelope;
+    }
```

Comment:
In #32904, we have this same need and we're doing it with an `addStamps()` method on `AbstractWorkerMessageEvent`. I believe we should use that approach. If you agree, it means we should wait for that PR to be merged or (better) duplicate that functionality here exactly so that either PR can be merged first without conflicts (if the code is duplicated *exactly*).